### ISSUE TITLE
Bug: `--on-failure DELETE` causes `Replacement type updates not supported on stack with disable-rollback`

### ISSUE THREAD

**Issue by eugene-bright** [aws/aws-sam-cli#4585]
### Description:

We are running sam with option `--on-failure DELETE` to prevent rollback failures to prevent stack from being stuck in `ROLLBACK_COMPLETE` status (we prefer automatic stack deletion). But because of that the stack is being marked with property `"DisableRollback": true` and it causes the following errors down the road:

```
Replacement type updates not supported on stack with disable-rollback. 
```

I'm pretty sure there is a way to correctly implement `--on-failure DELETE` without setting `"DisableRollback": true` .

Related to: #4164

### Steps to reproduce:

1. `sam deploy --on-failure DELETE ...`
2. change the template in a way that causes resource replacement and repeat deployment

### Observed result:

The option `"DisableRollback": true`  is set.

### Expected result:

IMHO, the option `DisableRollback` mustn't be set. 

### Additional environment details

1. OS: x86_64.ubuntu.22 (GitHub CI runner)
3. `sam --version`: SAM CLI, version 1.70.0
4. AWS region: us-east-1

**Comment by torresxb1** [comment]
Thanks for bringing this to our attention! I was able to replicate this issue. Let me check with the team if this was intentional (i.e. if there is a reason or necessity for why we disable the rollback in this case), and whether we can adjust the behaviour without breaking customers.

**Comment by jfuss** [comment]
@eugene-bright I am looking into this issue and from what I understand, this is the correct behavior you are seeing.

So `--on-failure` is only for Stack Creation not on update [boto3 docs on create-stack](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation/client/create_stack.html). The docs for `on-failure` both in SAM CLI and CloudFormation state: " Provide an action to determine what will happen when a stack fails to create.". So `on-failure` does nothing on `update-stack`, which is what you are doing. 

In terms of the `DisableRollback`, CloudFormation blocks updates to replacement actions. You can disable this through `--disable-rollback`.

**Comment by eugene-bright** [comment]
The question I have is why `sam --on-failure DELETE` sets `DisableRollback` property? It looks irrelevant and harmful for my workflow.
Thanks for clarification!

**Comment by eugene-bright** [comment]
We don't use `samconfig.toml`.
My previous observations show that  `--on-failure DELETE` sets `"DisableRollback": true`

**Comment by jfuss** [comment]
Adding bug label back, as I missed some code paths. Still trying to understand why this is here at all

**Comment by jfuss** [comment]
Found my confusion: The link in https://github.com/aws/aws-sam-cli/issues/4585#issuecomment-1456712118 is for `sync` not `deploy`. That at least explains why I was lost in how this was happening

**Comment by jfuss** [comment]
@eugene-bright After talking with some team members, I had a misunderstanding. I thought `on-failure` was something we just pass along to the CloudFormation APIs, since it wasn't supported in the APIs we call. I have come to learn that, we are trying to apply some logic to enable something similar to the `create/execute-changeset` calls we make. 

For now, the way to unblock yourself is to not provide `--on-failure`. This will leave the stacks up (won't be deleted), but will still allow for `disable-rollback` to be set. I will be working on trying to understand what the right thing here is and will keep the issue updated as I go.

**Comment by eugene-bright** [comment]
IMHO, the right thing is not to set `DisableRollback` when the stack is created with `--on-failure DELETE`. I don't want `DisableRollback` property to be set and it also looks completely irrelevant to `--on-failure DELETE` functionality.

**Comment by hawflau** [comment]
Closing as the fix has been released in AWS SAM CLI v1.77.0
